File: Coursera Courses/Specializations/Programming in Java: A Hands-on Introduction/Object-Oriented Java: Inheritance and Encapsulation/Week 3/LabChallenge.java
class Chef {
  private String name;
  private String cuisine;
  private int michelinStars;
  
  //add constructors below this line
  
  public Chef(String n){
    name = n;
  }
  
  public Chef(String n, String c){
    name = n;
    cuisine = c;
  }
  
  public Chef(String n, String c, int m){
    name = n;
    cuisine = c;
    michelinStars = m;
  }
  
  //add constructors above this line
  
  public String display() {
    String msg = String.format("%s is known for %s cuisine and has %d Michelin stars.", name, cuisine, michelinStars);
    return msg;
  }
}

public class LabChallenge {  
  public static void main(String[] args) {
    
    //add code below this line

    Chef c1 = new Chef("Marco Pierre White");
    Chef c2 = new Chef("Rene Redzepi", "Nordic");
    Chef c3 = new Chef("Thomas Keller", "French", 3);
    
    System.out.println(c1.display());
    System.out.println(c2.display());
    System.out.println(c3.display());
    
    //add code above this line
  }
}

File: Coursera Courses/Specializations/Programming in Python: A Hands-on Introduction/Object-Oriented Python: Inheritance and Encapsulation/Week 3/lab_challenge.py
class Chef:
  def __init__(self, name, cuisine, stars):
    self.name = name
    self.cuisine = cuisine
    self.michelin_stars = stars
    
  def __gt__(self, other_chef):
    return self.michelin_stars > other_chef.michelin_stars
    
  def compare(self, other_chef):
    if self > other_chef:
      return f'{self.name} has more Michelin stars than {other_chef.name}'
    else:
      return f'{other_chef.name} has more Michelin stars than {self.name}'
